device/api/umd/device/tt_soc_descriptor.h

```diff
@@ -149,14 +151,16 @@ class tt_SocDescriptor {
     void get_cores_and_grid_size_from_coordinate_manager();
 
     static tt_xy_pair calculate_grid_size(const std::vector<tt_xy_pair> &cores);
+    std::vector<tt::umd::CoreCoord> translate_coordinates(
+        const std::vector<tt::umd::CoreCoord> &physical_cores, const CoordSystem coord_system) const;
 
     // TODO: change this to unique pointer as soon as copying of tt_SocDescriptor
     // is not needed anymore. Soc descriptor and coordinate manager should be
     // created once per chip.
     std::shared_ptr<CoordinateManager> coordinate_manager = nullptr;
-    std::map<CoreType, std::vector<tt::umd::CoreCoord>> cores_map;
+    mutable std::map<CoreType, std::vector<tt::umd::CoreCoord>> cores_map;
```

Comment:
You added this because this is being used in some const function somewhere? We should ban this modifier.
If it is too much work to do the necessary changes, feel free to leave it for some later PR. But we should avoid this keyword.

Reply:
Added topic to https://github.com/tenstorrent/tt-umd/issues/356